src/actions/scanner.js

```diff
@@ -90,18 +90,26 @@ export function scannerDispatch (data) {
         let hash = await keccak(txRequest.data.rlp)
         dispatch(selectAccount(account))
         dispatch(scannedTx(hash, tx))
+        Actions.txDetails()
+        dispatch(resetScanner())
       } else if (txRequest.action === 'signTransactionHash') {
         var details = txRequest.data.details
         details.isSafe = false
         let hash = txRequest.data.hash
         dispatch(selectAccount(account))
         dispatch(scannedTx(hash, details))
+        Actions.txDetails()
+        dispatch(resetScanner())
+      } else if (txRequest.action === 'signData') {
+        let hash = txRequest.data.hash
```

Comment:
But the whole point is that you should be able to see the data before you sign it.
I can prepend the text in the UI (before constructing the QR code), but that could be done here as well (you can even re-use the rust function maybe if it's extracted to `ethkey/ethstore`)